### PR TITLE
feat(extension): Basic scene support

### PR DESCRIPTION
**Pull request by keiya01**
NOTE: To work this PR, this PR is necessary. https://github.com/reearth/reearth/pull/663

I added a basic scene support.
- Same view with VIEW3.0 prototype
  - The `map` type base-map is an alternative. I will update this after the same one with prototype is implemented.
- Share scene feature
  - You can test the share feature by adding `?share=123`.
Screenshot 2023-09-06 at 9 09 36

### CODE REVIEW

File: extension/src/shared/states/scene.ts [hunk 1]
```diff
@@ -1,6 +1,12 @@
-import { SceneProperty } from "../reearth/types";
-import { sharedAtom } from "../sharedAtoms";
+import { environmentTypeAtom } from "../../prototypes/view/states/app";
+import { graphicsQualityAtom } from "../../prototypes/view/states/graphics";
+import { sharedStoreAtomWrapper } from "../sharedAtoms";
 
-const initialScene: SceneProperty = {};
-
-export const sceneAtom = sharedAtom("scene", initialScene);
+export const sharableGraphicsQualityAtom = sharedStoreAtomWrapper(
```
Comment (KaWaite) [inline extension/src/shared/states/scene.ts:5]: shareable

File: extension/src/shared/states/scene.ts [hunk 2]
```diff
@@ -1,6 +1,12 @@
...
+  "graphicsQuality",
+  graphicsQualityAtom,
+);
+export const sharableEnvironmentTypeAtom = sharedStoreAtomWrapper(
```
Comment (KaWaite) [inline extension/src/shared/states/scene.ts:9]: shareable

**Review by KaWaite — APPROVED** [review]

File: extension/src/prototypes/view/environments/MapEnvironment.tsx [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+      lightIntensity={10}
+      shadowDarkness={colorMode === "light" ? 0.7 : 0.3}
+      imageBasedLightingIntensity={1}
+      sphericalHarmonicCoefficients={sphericalHarmonicCoefficients}
+      showSkyBox={false}
+      atmosphereSaturationShift={-1}
+      groundAtmosphereBrightnessShift={2}
+      // TODO(ReEarth): Use Takram's tile
+      // TODO(ReEarth): Support tile brightness
+      tiles={[{ id: "default", tile_type: "stamen_toner" }]}
```
Comment (airslice) [inline extension/src/prototypes/view/environments/MapEnvironment.tsx:53]: I think the array need to be wrapped with useMemo, it's causing update on each frame now.

File: extension/src/prototypes/view/environments/SatelliteEnvironment.tsx [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+  // }, [layer]);
+
+  return (
+    <Scene
+      enableGlobeLighting={enableTerrainLighting}
+      lightIntensity={1}
+      globeImageBasedLightingFactor={0.8}
+      shadowDarkness={0.5}
+      sphericalHarmonicCoefficients={sphericalHarmonicCoefficients}
+      tiles={[
```
Comment (airslice) [inline extension/src/prototypes/view/environments/SatelliteEnvironment.tsx:37]: The same as above.

**Review by airslice — APPROVED** [review]

**Pull request merged**